lldb/source/Commands/CommandObjectSource.cpp

```diff
@@ -1170,8 +1171,37 @@ class CommandObjectSourceList : public CommandObjectParsed {
           if (m_options.num_lines == 0)
             m_options.num_lines = 10;
           const uint32_t column = 0;
+
+          // Headers aren't always in the DWARF but if they have
+          // executable code (eg., inlined-functions) then the callsite's file(s)
+          // will be found.
+          // So if a header was requested and we got a primary file, then look
+          // thru its support file(s) for the header.
+          lldb::SupportFileSP actual_file_sp =
+              sc.comp_unit->GetPrimarySupportFile();
+          if (llvm::StringRef(m_options.file_name).ends_with(".h")) {
```

Comment:
There are many other "header file" extensions: `.H`, `.hpp`, `.hh`, etc. Also `.inc`  is used sometimes.

The standard headers don't have any extension at all: `<algorithm>` etc.

Is there a way to fix this without hard-coding one particular extension?

Reply:
How about we change to check to `if the file that was found has different extension from the requested`?
(That should cover most cases?)

Reply:
done!